src/lib/getElementType.js

```diff
@@ -19,7 +21,7 @@ function getElementType(Component, props, getDefault) {
   // computed default element type
 
   if (getDefault) {
-    const computedDefault = getDefault()
+    const computedDefault = typeof getDefault === 'function' ? getDefault() : getDefault
```

Comment:
If this change is made, the signature needs to be updated as well.

``jsx
 * @param {function} [getDefault] A function that returns a default element type.
```

That said, I'm not sure there is too much advantage to allowing a conditional here.

Reply:
Yes, I used `original` as you wrote below, so this change was reverted.